src/java.base/share/classes/java/lang/foreign/MemoryLayout.java

```diff
@@ -571,160 +673,4 @@ static PathElement sequenceElement() {
-     * object is also a layout, and it is equal to this layout. Two layouts are considered equal if they are of
-     * the same kind, have the same size, name and alignment constraints. Furthermore, depending on the layout kind, additional
-     * conditions must be satisfied:
-     * <ul>
-     *     <li>two value layouts are considered equal if they have the same {@linkplain ValueLayout#order() order},
-     *     and {@linkplain ValueLayout#carrier() carrier}</li>
-     *     <li>two sequence layouts are considered equal if they have the same element count (see {@link SequenceLayout#elementCount()}), and
-     *     if their element layouts (see {@link SequenceLayout#elementLayout()}) are also equal</li>
-     *     <li>two group layouts are considered equal if they are of the same type (see {@link StructLayout},
-     *     {@link UnionLayout}) and if their member layouts (see {@link GroupLayout#memberLayouts()}) are also equal</li>
-     * </ul>
-     *
-     * @param other the object to be compared for equality with this layout.
-     * @return {@code true} if the specified object is equal to this layout.
-     */
-    boolean equals(Object other);
-
-    /**
-     * {@return the hash code value for this layout}
-     */
-    int hashCode();
-
-    /**
-     * {@return the string representation of this layout}
-     */
-    @Override
-    String toString();
-
-    /**
-     * Creates a padding layout with the given size.
-     *
-     * @param size the padding size in bits.
-     * @return the new selector layout.
-     * @throws IllegalArgumentException if {@code size <= 0}.
-     */
-    static PaddingLayout paddingLayout(long size) {
-        MemoryLayoutUtil.checkSize(size);
-        return PaddingLayoutImpl.of(size);
-    }
-
-    /**
-     * Creates a value layout of given Java carrier and byte order. The type of resulting value layout is determined
-     * by the carrier provided:
-     * <ul>
-     *     <li>{@link ValueLayout.OfBoolean}, for {@code boolean.class}</li>
-     *     <li>{@link ValueLayout.OfByte}, for {@code byte.class}</li>
-     *     <li>{@link ValueLayout.OfShort}, for {@code short.class}</li>
-     *     <li>{@link ValueLayout.OfChar}, for {@code char.class}</li>
-     *     <li>{@link ValueLayout.OfInt}, for {@code int.class}</li>
-     *     <li>{@link ValueLayout.OfFloat}, for {@code float.class}</li>
-     *     <li>{@link ValueLayout.OfLong}, for {@code long.class}</li>
-     *     <li>{@link ValueLayout.OfDouble}, for {@code double.class}</li>
-     *     <li>{@link ValueLayout.OfAddress}, for {@code MemorySegment.class}</li>
-     * </ul>
-     * @param carrier the value layout carrier.
-     * @param order the value layout's byte order.
-     * @return a value layout with the given Java carrier and byte-order.
-     * @throws IllegalArgumentException if the carrier type is not supported.
-     */
-    static ValueLayout valueLayout(Class<?> carrier, ByteOrder order) {
```

Comment:
This method should not be removed. It is very useful when working with layouts programmatically (e.g. when a client wants to create a value layout from a class carrier). Even if it's unused in code (but it is used in one test), we should not treat this factory as useless. Also, I was only able to find this by looking at commits one by one. So, as a general comment, perhaps it would be a good idea to split the various changes.